AlphaWallet/WalletConnect/WalletConnectProvider.swift

```diff
@@ -299,12 +313,18 @@ extension WalletConnectProvider: WalletConnectServerDelegate {
             .mapError { WalletConnectError(error: $0) }
             .eraseToAnyPublisher()
         case .signEip712v3And4(let typedData):
-            return dappRequestProvider.requestSignMessage(
-                message: .eip712v3And4(typedData),
-                account: walletSession.account.address,
-                requester: requester)
-            .mapError { WalletConnectError(error: $0) }
-            .eraseToAnyPublisher()
+            do {
+                //TODO: maybe move varifying to sing message coordinator?
+                let typedData = try verifyIfEip712v3And4(typedData, walletSession: walletSession)
+                return dappRequestProvider.requestSignMessage(
+                    message: .eip712v3And4(typedData),
+                    account: walletSession.account.address,
+                    requester: requester)
+                .mapError { WalletConnectError(error: $0) }
+                .eraseToAnyPublisher()
+            } catch let e as WalletConnectError {
+                return .fail(e)
+            } catch { /*no-op*/ fatalError("inaccessible code path") }
         case .typedMessage(let typedData):
```

Comment:
Can you help check if the old EIP-712 (which is what `.typedMessage` is for) has a chain ID too? (also for dapp browser handling)